examples/openai-agents/python/main.py

```
@@ -0,0 +1,109 @@
+
+    Returns:
+        The agent's response as a string.
+    """
+    ctx = HonchoContext(user_id=user_id, session_id=session_id)
+
+    # Persist user message before the agent runs so it's available in context
+    save_memory(user_id, message, "user", session_id)
+
+    # Pass structured OpenAI-format history directly — don't flatten to plain text
+    history = get_context(ctx, tokens=2000)
+    input_messages = history + [{"role": "user", "content": message}]
```

Ruff (0.15.10)

[warning] Consider `[*history, {"role": "user", "content": message}]` instead of concatenation

Replace with `[*history, {"role": "user", "content": message}]`

(RUF005)